mxcubeweb/core/components/user/usermanager.py

```diff
@@ -131,10 +156,18 @@ def is_inhouse_user(self, user_id):
         return user_id in user_id_list
 
     # Abstract method to be implemented by concrete implementation
-    def _login(self, login_id, password):
+    def _login(self, login_id: str, password: str):
         pass
 
-    def login(self, login_id: str, password: str):
+    def login(self, login_id: str, password: str) -> None:
+        """
+        Create new session for the user if it does not exist. Activate user in
+        data store. If a sample is loaded in sample changer but not mounted,
+        mount it and update the smaple list. Try update the operator.
+
+        :param str login_id: username
+        :param str password: password
```

Comment:
We use Google style for docstrings:

* <https://mxcubeweb.readthedocs.io/en/latest/dev/contributing.html#style-guidelines>
* <https://www.sphinx-doc.org/en/master/usage/extensions/example_google.html>
* <https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html#google-vs-numpy>
*

Reply:
ok